fhir-persistence-schema/src/main/java/com/ibm/fhir/schema/app/SimpleSchemaPrinter.java

```diff
@@ -6,32 +6,3565 @@
+        // clear it out. 
+        commands.clear();
+        model.applyGrants(adapter, FhirSchemaConstants.FHIR_USER_GRANT_GROUP, "FHIRUSER");
+    }
+    
+    /**
+     * prints the grants
+     */
+    public void printGrants() {
+        printHeader(outGrants);
+        commands.keySet().stream().map(sql -> sql + DELIMITER + "\n").forEach(out::println);
+    }
+    
+    /*
+     * print solid line
+     */
+    private void printLine(PrintStream output) {
+        output.println("-------------------------------------------------------------------------------");
+    }
+    
+    /*
+     * print header
+     */
+    private void printHeader(PrintStream output) {
+        int year = java.time.ZonedDateTime.now(ZoneId.of("UTC")).get(ChronoField.YEAR);
+        printLine(output);
+        output.println("-- (C) Copyright IBM Corp. 2016, " + year);
+        output.println("--");
+        output.println("-- SPDX-License-Identifier: Apache-2.0");
+        printLine(output);
+        output.println("-- IBM FHIR Server - Schema Output");
+        output.println("-- Generated at " + Instant.now() + "");
+        printLine(output);
+        output.println("");
+    }
+    /**
+     * prints the schema out to two locations (sometimes the same printstream
+     */
+    public void print() {
+        printHeader(out);
+        out.println("CREATE SCHEMA " + SCHEMA_NAME + DELIMITER + "\n");
+        out.println("CREATE SCHEMA " + ADMIN_SCHEMA_NAME + DELIMITER + "\n");
+        commands.keySet().stream().map(sql -> sql + DELIMITER + "\n").forEach(out::println);
         
-        // Pretend that our target is a DB2 database
-        Db2Adapter adapter = new Db2Adapter(tgt);
-        model.apply(adapter);
+        printHeader(outStoredProcedure);
+        storedProceduresCommands.keySet().stream().map(sql -> sql + STORED_PROCEDURE_DELIMITER + "\n").forEach(outStoredProcedure::println);
+
+        // Intentionally DO NOT close the streams. 
+    }
+
+
+    public static void main(String[] args) {
+
+        boolean outputToFile = false;
+        String outputFile = "";
+
+        // If there are files
+        for (int i = 0; i < args.length; i++) {
+            String arg = args[i];
+            switch (arg) {
+            case "--to-file":
+                outputToFile = true;
+                break;
+            default:
+                throw new IllegalArgumentException("Please use the --output");
+            }
+        }
+
+        try {
+            SimpleSchemaPrinter printer = new SimpleSchemaPrinter(outputToFile);
+            printer.process();
+            printer.print();
+            printer.processApplyGrants();
+            printer.printGrants();
+        } catch (FileNotFoundException e) {
+            System.out.println("The File is not able to be created or found - [" + outputFile + "]");
+        }
+    }
+
+    /*
+     * The following classes are 'dummy implementations' that enable printing of the
+     * SQL.
```

Comment:
why do we need the dummy stubs?  no way to just use the existing Print classes?
if its really needed, maybe we introduce test dependency for creating the stubs from the Interface?

Reply:
existing print classes miss many commands and do not fully include all the commands

Reply:
can we fix those?

Reply:
Not really.  It'll be a pain to re-engineer to support it; I looked at it.  This is the reliable way - regardless of IDatabaseTarget and IDatabaseTranslator.